File: arlo-admin-server/internal/modules/message/model/notice.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Notice 通知公告模型，对应 sys_notice 表
type Notice struct {
	ID          uint64         `gorm:"primaryKey" json:"id"`
	Title       string         `gorm:"size:128;not null;default:''" json:"title"`
	Content     string         `gorm:"type:text" json:"content"`
	Type        int8           `gorm:"not null;default:1" json:"type"`      // 1=通知, 2=公告
	Level       int8           `gorm:"not null;default:1" json:"level"`     // 1=普通, 2=重要, 3=紧急
	Status      int8           `gorm:"not null;default:0" json:"status"`    // 0=草稿, 1=已发布, 2=已撤回
	PublisherID uint64         `gorm:"not null;default:0" json:"publisherId"`
	Publisher   string         `gorm:"size:32;not null;default:''" json:"publisher"`
	CreatedAt   time.Time      `json:"createdAt"`
	UpdatedAt   time.Time      `json:"updatedAt"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

func (Notice) TableName() string {
	return "sys_notice"
}

File: arlo-admin-web/src/api/modules/notice.ts
import request from '../request'

// ===================== 通知公告 =====================
export interface NoticeItem {
  id: number
  title: string
  content: string
  type: number        // 1=通知 2=公告
  level: number       // 1=普通 2=重要 3=紧急
  status: number      // 0=草稿 1=已发布 2=已撤回
  publisherId: number
  publisher: string
  createdAt: string
  updatedAt: string
}

export interface NoticePageResult {
  list: NoticeItem[]
  total: number
  page: number
  pageSize: number
}

export interface NoticeListQuery {
  title?: string
  status?: number
  type?: number
  page: number
  pageSize: number
}

export interface NoticeFormParams {
  title: string
  content: string
  type: number
  level: number
}

const BASE = '/v1/message/notice'

export function getNoticeList(params: NoticeListQuery) {
  return request.get<NoticePageResult>(`${BASE}/list`, params)
}

export function getNoticeDetail(id: number) {
  return request.get<NoticeItem>(`${BASE}/${id}`)
}

export function createNotice(data: NoticeFormParams) {
  return request.post<NoticeItem>(BASE, data)
}

export function updateNotice(id: number, data: NoticeFormParams) {
  return request.put<NoticeItem>(`${BASE}/${id}`, data)
}

export function deleteNotice(id: number) {
  return request.delete(`${BASE}/${id}`)
}

export function publishNotice(id: number) {
  return request.put(`${BASE}/${id}/publish`)
}

export function revokeNotice(id: number) {
  return request.put(`${BASE}/${id}/revoke`)
}
